kythe/go/extractors/validation/validation.go

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package validation contains logic for verifing the contents of a kzip.
+package validation
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"kythe.io/kythe/go/platform/kzip"
+)
+
+// Harness encapsulates data for validation of a single kzip and repo.
+type Harness struct {
+	// Path to the kzip file to validate.
+	Kzip string
```

Comment:
This is spelled `KZIP` elsewhere in go.

Can we make this more generic, though?  Just `ArchivePath` or something?  Or maybe `Compilations []string` and allow multiple `.kzip` files?